### PR TITLE
[fix-#13045] fix endless loop

### PR DESCRIPTION
**Pull request by fuchanghai**
## Purpose of the pull request

This PR is complementary to issue #13045

PR #13051

## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by fuchanghai** [comment]
@zhuangchong PTAL. cc @huage1994

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        WorkflowStateEvent workflowStateEvent = (WorkflowStateEvent) stateEvent;
+        ProcessInstance processInstance =
+                workflowExecuteRunnable.getWorkflowExecuteContext().getWorkflowInstance();
+        measureProcessState(workflowStateEvent, processInstance.getProcessDefinitionCode().toString());
+        log.info(
+                "Handle workflow instance submit fail state event, the current workflow instance state {} will be changed to {}",
+                processInstance.getState(), workflowStateEvent.getStatus());
+
+        if (workflowExecuteRunnable.processComplementData()) {
+            return true;
```
Comment (EricGao888) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java]: Will this patch fix the endless loop when completing data?

Comment (fuchanghai) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java]: > Will this patch fix the endless loop when completing data?

yes ,in #13051 ,forgot to add the corresponding event processing class, resulting in the status of the process instance not being updated

**Comment by fuchanghai** [comment]
@huage1994 PTAL  cc @ruanwenjun

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java (hunk 1 above)
Comment (EricGao888) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java]: I'm still a little confused here. If a workflow instance fails to get submitted in `data complement mode`, in line 45, we directly `return true` without doing anything else to it. What will the state of the workflow instance become? Failure or just hanging there?

**Review by EricGao888 — COMMENTED** [review]
BTW, you need to change the log information here:
https://github.com/apache/dolphinscheduler/blob/8c01fcf14b199e2c1ba86c17351e1ea30614f484/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowStartEventHandler.java#L70-L79

If DS fails to submit a workflow instance, it no more resends the start event.

**Comment by fuchanghai** [comment]
> BTW, you need to change the log information here:
> 
> https://github.com/apache/dolphinscheduler/blob/8c01fcf14b199e2c1ba86c17351e1ea30614f484/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowStartEventHandler.java#L70-L79
> 
> If DS fails to submit a workflow instance, it no more resends the start event.

done

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java (hunk 1 above)
Comment (fuchanghai) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/WorkflowSubmitFailStateEventHandler.java]: > I'm still a little confused here. If a workflow instance fails to get submitted in `data complement mode`, in line 45, we directly `return true` without doing anything else to it. What will the state of the workflow instance become? Failure or just hanging there?

You are right, whatever type, as long as an error occurs during the submission phase, the status should be updated to failure

**Review by EricGao888 — APPROVED** [review]

**Review by huage1994 — APPROVED** [review]
LGTM

**Pull request merged**